modules/client/src/controllers/ResolveConditionController.ts

```diff
@@ -33,23 +35,25 @@ export class ResolveConditionController extends AbstractController {
   private resolveLinkedTransfer = async (
     params: ResolveLinkedTransferParameters,
   ): Promise<ResolveLinkedTransferResponse> => {
+    this.log.info(`Resolving link: ${JSON.stringify(params)}`);
+
     // convert and validate
-    const { paymentId, preImage, amount, assetId } = convert.ResolveLinkedTransfer(
-      "bignumber",
-      params,
-    );
+    const { assetId, amount } = await this.node.fetchLinkedTransfer(params.paymentId);
+    const amountBN = bigNumberify(amount);
+    this.log.info(`Found link payment for ${amountBN.toString()} ${assetId}`);
 
     const freeBal = await this.connext.getFreeBalance(assetId);
     const preTransferBal = freeBal[this.connext.freeBalanceAddress];
 
     // TODO: dont listen to linked transfer app in default listener, only listen for it here
 
-    await this.node.resolveLinkedTransfer(paymentId, preImage, amount, assetId);
+    const { paymentId, preImage } = params;
+    await this.node.resolveLinkedTransfer(paymentId, preImage);
```

Comment:
The node has info in it's db that links paymentId to the proper amount & assetId so no need for the client to provide either of these derived fields. This also eliminates complexity surrounding case where a user submits mismatched paymentId/amount/assetId